source/whatever/index.html

```diff
@@ -28,35 +28,31 @@ <h3>Module Management</h3>
-            <li><a
-                href="https://github.com/jrockway/cperl-mode">Unified
-                Perl 6 + Perl + Moose/MooseX::Declare Emacs syntax
-                highlighting</a></li>
-	    <li><a href="https://github.com/syl20bnr/spacemacs">Spacemacs</a>,
-		    an Emacs wrapper with vim key-bindings and with extra stuff.</li>
-	    <li><a href="https://github.com/hankache/perl6.nanorc">perl6.nanorc</a>, anyone using nano as his/her text editor would be able to use it to get syntax highlighting for Perl&nbsp;6</li>
-	    <li>
-                <a href="http://uvviewsoft.com/cudatext/">CudaText</a> has Perl 6 lexer in Addon Manager
-	    </li>
+          <li><a href="https://commaide.com/">CommaIDE</a>: designed specifically for Perl 6.
+              Community edition available for free, additional features and more regular updates through paid Supporter Program.</li>
+          <li><a href="https://atom.io">Atom</a>:
+            <ul>
+              <li><a href="https://atom.io/packages/language-perl6">Perl 6 syntax highlighter</a>,
+                  a plugin that provides Perl 6 highlighting. Follow the instructions in the package readme.</li>
+              <li><a href="https://atom.io/packages/script">Perl 6 script runner</a></li>
+              <li>See here how to configure both the above plugins, as well as others,
+                  to turn <a href="https://github.com/perl6/Atom-as-a-Perl6-IDE">Atom into a Perl 6 IDE</a>.
+                  Works with the syntax highlighter listed above. Can run Perl 6 code from the screen buffer, selection, or a file.</li>
+              <li><a href="https://atom.io/packages/atom-perl6-editor-tools">Perl 6 Editor Tools</a>, an Atom plugin that
+                  provides a collection of useful Perl 6 editing tools (may not be compatible with the latest Atom).</li>
+            </ul>
+          <li><a href="http://padre.perlide.org/">Padre, the Perl IDE</a>:
+              <a href="https://metacpan.org/pod/Padre::Plugin::Perl6">Padre::Plugin::Perl6</a>, Perl 6 highlighting and code execution</li>
+          <li>Vim: <a href="https://github.com/vim-perl/vim-perl6">Perl 6 syntax highlighting</a> for better highlighting</li>
+          <li><a href="https://neovim.io/">Neovim</a>: <a href="https://github.com/vim-perl/vim-perl6">vim-perl6 plugin</a> for better highlighting</li>
+          <li>Emacs:
+            <ul>
+              <li><a href="https://github.com/perl6/perl6-mode">perl6-mode</a>, an Emacs major mode for Perl 6 which provides syntax highlighting (and more)</li>
+              <li><a href="https://github.com/jrockway/cperl-mode">Unified Perl 6 + Perl + Moose/MooseX::Declare</a> syntax highlighting</li>
+              <li><a href="https://github.com/syl20bnr/spacemacs">Spacemacs</a>: Emacs wrapper with vim key-bindings and with extra stuff</li>
+            </ul>
+          <li>Nano: <a href="https://github.com/hankache/perl6.nanorc">Perl 6 syntax highlighting</a></li>
```

Comment:
Missing closing </li> here? Can you please lint or check in the browser the HTML and correct errors?

Reply:
The <li> for Emacs is closed at the <ul> because it’s a block element. It’s valid html5.